src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/ProductOfConditions.hpp

```diff
@@ -23,12 +24,171 @@
 #include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
 #include "Options/Options.hpp"
 #include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Lapse.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Shift.hpp"
+#include "PointwiseFunctions/GeneralRelativity/SpatialMetric.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
 namespace grmhd::GhValenciaDivClean::BoundaryConditions {
 namespace detail {
+template <evolution::BoundaryConditions::Type GhBcType,
+          evolution::BoundaryConditions::Type ValenciaBcType>
+struct UnionOfBcTypes {
+  static constexpr evolution::BoundaryConditions::Type bc_type =
+      evolution::BoundaryConditions::Type::GhostAndTimeDerivative;
+};
+
+template <>
+struct UnionOfBcTypes<evolution::BoundaryConditions::Type::Ghost,
```

Comment:
The only downside with that strategy is that I wouldn't be able to use the `static_assert`s for the outflow condition the same way to avoid an ambiguous specialization -- I'd either need SFINAE to select only the case where one is outflow and the other is not, or I'd need to specify all of the combinations with one outflow condition and one non-outflow, which would be a larger number of individual specialization cases.

Reply:
ah yea, that makes sense. Thanks :)